candle-pyo3/src/lib.rs

```diff
@@ -772,6 +772,39 @@ impl PyTensor {
         Ok(PyTensor(self.0.copy().map_err(wrap_err)?))
     }
 
+    #[pyo3(signature = (*args, **kwargs), text_signature = "(self, *args, **kwargs)")]
+    /// Performs Tensor dtype and/or device conversion.
+    /// &RETURNS&: Tensor
+    fn to(&self, args: &PyTuple, kwargs: Option<&PyDict>) -> PyResult<Self> {
+        let mut t = self.0.clone();
+        //handle args
+        if !args.is_empty() {
+            for arg in args.iter() {
```

Comment:
I don't think we should process the arguments in order. If there are multiple device arguments for example the behavior would be really weird and I don't see a legitimate use case for this.

Reply:
Yup i will enforce that only one dtype or device can be provided via the args or kwargs and return an error if multiple were provided.